web_src/js/components/RepoBranchTagSelector.vue

```diff
@@ -30,6 +30,10 @@
         <div class="loading-indicator is-loading" v-if="isLoading"/>
         <div v-for="(item, index) in filteredItems" :key="item.name" class="item" :class="{selected: item.selected, active: active === index}" @click="selectItem(item)" :ref="'listItem' + index">
           {{ item.name }}
+          <!-- in backend, we put default branch as the first item, so just check item index is enought here -->
+          <div class="ui label" v-if="index===0 && mode === 'branches'">
```

Comment:
This assumption is too fragile, IMO the right approach is let the backend tell frontend which is the default branch.

Reply:
Done.